numpyro/distributions/constraints.py

```diff
@@ -638,6 +639,11 @@ def feasible_like(self, prototype):
         return jax.numpy.zeros_like(prototype)
 
 
+class _Real(_Complex):
+    def __call__(self, x):
+        return super().__call__(x) & (jax.numpy.isreal(x))
```

Comment:
Surprisingly, adding the `&` leads to some warnings *not* being emitted in `test/test_distributions.py::test_distribution_constraints`, but only when `jit`-ed.

Reply:
I think `isreal` is a jax function, which will generate a tracer under `jit`.

One option is to switch to np like in PositiveDefinite. It is better to have separate instances for real and complex though. I don't think that current constraints work for complex numbers.